ui/app/AppLayouts/Browser/JSDialogWindow.qml

```diff
@@ -0,0 +1,84 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtWebEngine 1.10
+import "../../../shared"
+import "../../../imports"
+
+ModalPopup {
+    id: root
+    property QtObject request
+    height: 286
+
+    closePolicy: Popup.NoAutoClose
+
+    onClosed: {
+        request.dialogReject();
+        root.destroy();
+    }
+
+    Component.onCompleted: {
+        root.title = request.securityOrigin;
+        message.text = request.message;
+        if(request.type == JavaScriptDialogRequest.DialogTypeAlert){
+            cancelButton.visible = false;
+        }
+        if(request.type == JavaScriptDialogRequest.DialogTypePrompt){
+            prompt.text = request.defaultText;
+            prompt.visible = true;
+            svMessage.height = 75;
+        }
+    }
+
+    ScrollView {
+        id: svMessage
+        width: parent.width
+        height: 100
+        TextArea {
+            id: message
+            wrapMode: TextEdit.Wrap
+            readOnly: true
+            text: ""
+        }
+    }
+
+    Input {
+        id: prompt
+        text: ""
+        visible: false
+        Layout.fillWidth: true
+        anchors.top: svMessage.bottom
+    }
+
+    footer: Item {
+        anchors.top: parent.bottom
+        anchors.right: parent.right
+        anchors.bottom: root.bottom
+        anchors.left: parent.left
+
+        StyledButton {
+            id: okButton
+            anchors.right: parent.right
+            anchors.rightMargin: Style.current.smallPadding
```

Comment:
You shouldn't need the margin. The parent should have the necessary padding already